### PR TITLE
 BUG: Unwanted conversion from timedelta to float (#18493)

### LINKED ISSUE #18493: Unwanted conversion from timedelta to float (but not datetime)
In the code below, masked assignment of a timedelta series converts the series values to float64. The same code running against a datetime correctly keeps the series dtype.

    In [1]: import pandas as pd

    In [2]: dseries=pd.Series(pd.date_range('1/1/2011', periods=5, freq='H')); dseries=dseries-dseries[0]; dseries
    Out[2]: 
    0   00:00:00
    1   01:00:00
    2   02:00:00
    3   03:00:00
    4   04:00:00
    dtype: timedelta64[ns]

    In [3]: dseries[dseries==dseries[0]]=None; dseries
    Out[3]: 
    0   -9.223372e+18
    1    3.600000e+12
    2    7.200000e+12
    3    1.080000e+13
    4    1.440000e+13
    dtype: float64

    In [4]: dseries=pd.Series(pd.date_range('1/1/2011', periods=5, freq='H')); dseries=dseries; dseries
    Out[4]: 
    0   2011-01-01 00:00:00
    1   2011-01-01 01:00:00
    2   2011-01-01 02:00:00
    3   2011-01-01 03:00:00
    4   2011-01-01 04:00:00
    dtype: datetime64[ns]

    In [5]: dseries[dseries==dseries[0]]=None; dseries
    Out[5]: 
    0                   NaT
    1   2011-01-01 01:00:00
    2   2011-01-01 02:00:00
    3   2011-01-01 03:00:00
    4   2011-01-01 04:00:00
    dtype: datetime64[ns]

### PR DESCRIPTION
**Pull request by fjdiod**
- [ ] closes #18493
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.22.0.txt [hunk 1]
```diff
@@ -151,7 +151,7 @@ Indexing
 - Bug in :class:`IntervalIndex` where empty and purely NA data was constructed inconsistently depending on the construction method (:issue:`18421`)
 - Bug in ``IntervalIndex.symmetric_difference()`` where the symmetric difference with a non-``IntervalIndex`` did not raise (:issue:`18475`)
 - Bug in indexing a datetimelike ``Index`` that raised ``ValueError`` instead of ``IndexError`` (:issue:`18386`).
-
+- Bug in ``TimeDeltaBlock._can_hold_element()`` where masked assignment of a timedelta series converts the series values to float64 (:issue:`18493`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: you can move to 0.21.1

don't use internal language here, these are user-level notes. So say something like

Bug in masked assignment of a ``timedelta64[ns]`` dtype ``Series`` with a ``Timedelta`` incorrectly coerced to float.

File: pandas/tests/indexing/test_timedelta.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 import pandas as pd
 from pandas.util import testing as tm
+from numpy import nan
```
Comment (jreback) [inline pandas/tests/indexing/test_timedelta.py]: don't import this, use ``np.nan``

File: pandas/tests/indexing/test_timedelta.py [hunk 3]
```diff
@@ -47,3 +48,14 @@ def test_string_indexing(self):
...
         sliced = df.loc['0 days']
         tm.assert_series_equal(sliced, expected)
+
+    @pytest.mark.parametrize(
+        "value, expected",
+        [(None, [pd.NaT, 1, 2]),
+         (pd.NaT, [pd.NaT, 1, 2]),
+         (nan, [pd.NaT, 1, 2])])
+    def test_nan(self, value, expected):
+        series = pd.Series([0, 1, 2], dtype='timedelta64[ns]')
```
Comment (jreback) [inline pandas/tests/indexing/test_timedelta.py]: add the issue number here as a commet

File: pandas/tests/indexing/test_timedelta.py [hunk 4]
```diff
@@ -47,3 +48,14 @@ def test_string_indexing(self):
         expected = df.iloc[0]
         sliced = df.loc['0 days']
         tm.assert_series_equal(sliced, expected)
+
+    @pytest.mark.parametrize(
+        "value, expected",
+        [(None, [pd.NaT, 1, 2]),
```
Comment (jreback) [inline pandas/tests/indexing/test_timedelta.py]: you don't need to pass expected here as they are the same, just define in the body

File: pandas/tests/indexing/test_timedelta.py [hunk 5]
```diff
@@ -47,3 +48,14 @@ def test_string_indexing(self):
...
+         (pd.NaT, [pd.NaT, 1, 2]),
+         (nan, [pd.NaT, 1, 2])])
+    def test_nan(self, value, expected):
```
Comment (jreback) [inline pandas/tests/indexing/test_timedelta.py]: name this like: test_masked_setitem

File: pandas/tests/indexing/test_timedelta.py [hunk 6]
```diff
@@ -47,3 +48,14 @@ def test_string_indexing(self):
...
+        series = pd.Series([0, 1, 2], dtype='timedelta64[ns]')
+        series[series == series[0]] = value
+        expected = pd.Series([pd.NaT, 1, 2], dtype='timedelta64[ns]')
```
Comment (jreback) [inline pandas/tests/indexing/test_timedelta.py]: test with ``.loc`` as well

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. mostly style comments. ping on green.

**Comment by fjdiod** [comment]
Hello, ok will fix that. Also, additional commits should be combined into one?

**Comment by jreback** [comment]
you don’t need to combine you can just push more

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @fjdiod

**Pull request merged**